packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/string_database/dart_string_database.web.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/index_db/index_db_string_db.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database_element.dart';
+import 'package:aws_common/aws_common.dart';
+import 'package:drift/drift.dart';
+
+/// {@macro amplify_analytics_pinpoint_dart.dart_string_database}
+class DartStringDatabase implements StringDatabase, Closeable {
```

Comment:
Move the logic from `IndexStringDb` to here. There's way too much indirection going on.